src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -81,7 +81,7 @@ The .NET Foundation licenses this file to you under the MIT license.
 
     <NativeObject>$(NativeIntermediateOutputPath)$(TargetName)$(NativeObjectExt)</NativeObject>
     <NativeBinary>$(NativeOutputPath)$(TargetName)$(NativeBinaryExt)</NativeBinary>
-    <IlcExportUnmanagedEntrypoints Condition="'$(IlcExportUnmanagedEntrypoints)' == '' and '$(NativeLib)' == 'Shared'">true</IlcExportUnmanagedEntrypoints>
+    <IlcExportUnmanagedEntrypoints Condition="'$(IlcExportUnmanagedEntrypoints)' == '' and ('$(NativeLib)' == 'Shared' or '$(NativeLib)' == 'Static')">true</IlcExportUnmanagedEntrypoints>
```

Comment:
When people build static libraries, they likely want the unmanaged entrypoints to be used internally within the final binary, without them being exported.

Reply:
This opt-in vs. opt-out. If user is going to link the static library to shared libraryy, then they may want to export the `UnmanagedCallersOnly(EntryPoint=..` APIs like we do in shared lib publishing. If they are going to use exported APIs in their glue code only and generate an executable, then they can opt-out by setting `IlcExportUnmanagedEntrypoints=false` or deleting `-Wl,--version-script..` line from `linkoptions.txt`.

Reply:
Yes, this is about defaults. I am arguing that `IlcExportUnmanagedEntrypoints=false` is a better default for static libraries, for similar reason that it is a better default for executables.

Reply:
If someone wants to mix nativeAOT'd code with C code and build a DLL, there are these options:

1. The native AOT implementations are the public interface they want to export from the DLL and C is only an implementation detail they call into from C#. It would be more natural to build as `NativeLib=Shared` and add whatever other C code they have in the form of .a files as additional input to PublishAot.
2. The public API of the DLL is in C and there's some implementation details in C#. They'd take the .a file from PublishAot and take it from there using non-.NET tooling.
3. There could also be something in between (some of the public APIs of the DLL are native AOT, some are in C), but I don't know if that's the thing we should optimize for.

I'd also lean towards `IlcExportUnmanagedEntrypoints=false` being the better default.